src/somnium/congomongo.clj

```diff
@@ -412,8 +412,8 @@ When with-mongo and set-connection! interact, last one wins"
   (when (and one? sort)
     (throw (IllegalArgumentException. "Fetch :one? (or fetch-one) can't be used with :sort.
 You should use fetch with :limit 1 instead."))); one? and sort should NEVER be called together
-  (when (and one? (or read-preferences (not= [] options) (not= 0 limit) hint explain?))
-    (throw (IllegalArgumentException. "At the moment, fetch-one doesn't support read-preferences, options, limit or hint"))) ;; these are allowed but not implemented here
+  (when (and one? (or (not= [] options) (not= 0 limit) hint explain?))
+    (throw (IllegalArgumentException. "At the moment, fetch-one doesn't support options, limit, or hint"))) ;; these are allowed but not implemented here
```

Comment:
~~~~ Might be useful to indicate that the exception is coming from congomongo rather than MongoDB itself -- I thought that the new mongo 3.0 removed support rather than congomongo.